integrationTests/chainSimulator/bridge/executeAndBurn_test.go

```diff
@@ -0,0 +1,308 @@
+)
+
+const (
+	issuePrice = "5000000000000000000"
+)
+
+func TestChainSimulator_ExecuteWithMintAndBurnFungibleWithDeposit(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	token := "sov1-SOVTKN-1a2b3c"
+	tokenNonce := uint64(0)
+
+	mintTokens := make([]chainSim.ArgsDepositToken, 0)
+	mintTokens = append(mintTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(123),
+		Type:       core.Fungible,
+	})
+	mintTokens = append(mintTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(100),
+		Type:       core.Fungible,
+	})
+
+	depositTokens := make([]chainSim.ArgsDepositToken, 0)
+	depositTokens = append(depositTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(12),
+		Type:       core.Fungible,
+	})
+	depositTokens = append(depositTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      tokenNonce,
+		Amount:     big.NewInt(10),
+		Type:       core.Fungible,
+	})
+
+	simulateExecutionAndDeposit(t, mintTokens, depositTokens)
+}
+
+func TestChainSimulator_ExecuteWithMintAndBurnNftWithDeposit(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	nft := "sov2-SOVNFT-123456"
+	nftNonce := uint64(1)
+	token := "sov3-TKN-1q2w3e"
+
+	mintTokens := make([]chainSim.ArgsDepositToken, 0)
+	mintTokens = append(mintTokens, chainSim.ArgsDepositToken{
+		Identifier: nft,
+		Nonce:      nftNonce,
+		Amount:     big.NewInt(1),
+		Type:       core.NonFungible,
+	})
+	mintTokens = append(mintTokens, chainSim.ArgsDepositToken{
+		Identifier: token,
+		Nonce:      0,
+		Amount:     big.NewInt(1),
+		Type:       core.Fungible,
+	})
+
+	depositTokens := make([]chainSim.ArgsDepositToken, 0)
+	depositTokens = append(depositTokens, chainSim.ArgsDepositToken{
+		Identifier: nft,
+		Nonce:      nftNonce,
+		Amount:     big.NewInt(1),
+		Type:       core.NonFungible,
+	})
+
+	simulateExecutionAndDeposit(t, mintTokens, depositTokens)
+}
+
+func TestChainSimulator_ExecuteWithMintAndBurnSftWithDeposit(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	sft := "sov3-SOVSFT-654321"
+	sftNonce := uint64(1)
+
+	mintTokens := make([]chainSim.ArgsDepositToken, 0)
+	mintTokens = append(mintTokens, chainSim.ArgsDepositToken{
+		Identifier: sft,
+		Nonce:      sftNonce,
+		Amount:     big.NewInt(50),
+		Type:       core.SemiFungible,
+	})
+
+	depositTokens := make([]chainSim.ArgsDepositToken, 0)
+	depositTokens = append(depositTokens, chainSim.ArgsDepositToken{
+		Identifier: sft,
+		Nonce:      sftNonce,
+		Amount:     big.NewInt(20),
+		Type:       core.SemiFungible,
+	})
+
+	simulateExecutionAndDeposit(t, mintTokens, depositTokens)
+}
+
+func simulateExecutionAndDeposit(
+	t *testing.T,
+	mintTokens []chainSim.ArgsDepositToken,
+	depositTokens []chainSim.ArgsDepositToken,
```

Comment:
I think we should find a better naming for these variables, here and everywhere in tests(also rename the tests-that execute is the endpoint of the contract, we are not testing that actually).

Something perhaps like bridgedIn/bridgedOut tokens?